.github/PULL_REQUEST_TEMPLATES/pre-release-nightly-demos.md

```diff
@@ -0,0 +1,53 @@
+# Run and Update Demos on Nightly for Stackable Release XX.(X)X
+
+<!--
+    Make sure to update the link in 'issues/.github/ISSUE_TEMPLATE/release.md'
+    when you rename this file.
+-->
+
+<!--
+    Replace 'TRACKING_ISSUE' with the applicable release tracking issue number.
+-->
+
+Part of <https://github.com/stackabletech/issues/issues/TRACKING_ISSUE>
+
+> [!NOTE]
+> Demos run on the latest stable release by default. This PR includes changes
+> to commands, documentation and UI screenshots when running the demos on
+> nightly instead. The recommended workflow looks like this:
+>
+> 1. Create a `.env` file in your local copy of this repository.
+> 2. Create local stack and demo definition files, like `stacks-v2.local.yaml`
+>    and `demos-v2.local.yaml`. They should use the nightly version instead of
+>    stable. They also might need to point to local manifests instead of remote
+>    ones.
+> 3. Add the `STACKABLE_STACK_FILES` and `STACKABLE_DEMO_FILES` env vars to the
+>    file. They should point to the absolute path of your local stack and demo
+>    definition files.
```

Comment:
_Just adding a different perspective to consider..._

Tbh, I prefer using the checked out stack files explicitly on the command line.
The same changes get made, as you described, but if there are actual fixes needed, it is a matter of editing the same files and only checking in those changes (and reverting the other ones that get it working with nightly).

Not at all suggesting my way is better, but for me, doing the above, I would forget to unset the variable, or I would run a new shell that doesn't have it set, and invalidate the testing). I use the shell history heavily.